### DISCUSSION TITLE
site2site

### DISCUSSION THREAD

**Discussion by helio58** [wg-easy/wg-easy#866] [Ideas]
Is possible to use wg-easy site2site ?
If so is there a guide?
Thanks for your work.

**Comment by peterlewis** [comment]
Interesting question, thank you. It's not the intended use case for wg-easy (as for a site-to-site you might as well set up a dedicated wg instance). As to whether or not it's possible, I do not know, as haven't tried, myself.

**Comment by helio58** [comment]
Thank  you Peter.
I m using as is intended. And it works very well.
I have a situation that the ideal should be site-to-site , i m a newbie to routing I have follow countless guides but haven t succed.
Once more thanks

**Comment by sudo-nitz** [comment]
I think wg-easy is strict to point-to-point architecture.

When building point-to-site or site-to-site You have to add site's subnet into AllowedIP parameter.

For example:
DEVICE A: Your first subnet is 192.168.1.0/24. There is a computer with Wireguard docker (wg-easy) with Wireguard address 10.8.0.1
DEVICE B: Your second subnet is 192.168.2.0/24. There is a computer with Wireguard client with Wireguard address 10.8.0.2.
DEVICE C: There is a third computer in internet with Wireguard address 10.8.0.3.

When You want to access LAN devices from second subnet from internet through Wireguard docker You have to add second subnet to AllowedIP for Wireguard address 10.8.0.1.

More details:
DEVICE A:
[Interface]
Address = 10.8.0.1/24
PrivateKey = <HOST 'A' PRIVATE KEY>
ListenPort = 51820

[Peer]
PublicKey = <HOST 'B' PUBLIC KEY>
AllowedIPs = 10.8.0.2/32, **192.168.2.0/24**

[Peer]
PublicKey = <HOST 'C' PUBLIC KEY>
AllowedIPs = 10.8.0.3/32

DEVICE B:
[Interface]
Address = 10.8.0.2/24                  
PrivateKey = <HOST 'B' PRIVATE-KEY>
ListenPort = 51820
DNS = 1.1.1.1  

[Peer]
PublicKey = <PUBLIC KEY OF HOST 'A'>
Endpoint = <DOMAIN>:51820
AllowedIPs = 0.0.0.0/0, ::/0

DEVICE C:
[Interface]
Address = 10.8.0.3/24                  
PrivateKey = <HOST 'C' PRIVATE-KEY>
ListenPort = 51820
DNS = 1.1.1.1  

[Peer]
PublicKey = <PUBLIC KEY OF HOST 'A'>
Endpoint = <DOMAIN>:51820
AllowedIPs = 0.0.0.0/0, ::/0

As You can see above, currently there is no option to add specific individual addresses for each peer in wg-easy.

**Comment by sudo-nitz** [comment]
Offcourse I'm wrong please correct me.

I'm also interested in accessing devices from client's subnet. I don't know how to do this with wg-easy.
